app/models/role.rb

```diff
@@ -0,0 +1,9 @@
+class Role < ApplicationRecord
+  has_and_belongs_to_many :users, :join_table => :users_roles
+  
+  belongs_to :resource, polymorphic: true, optional: true
+  
```

Comment:
File generated by Rolify.  Belongs to resource indicates the optional scalability of assigning polymorphic associations to a user depending if we want them to have controlled access to editing resources. (like creating a moderator role)